config/groovy/common.groovy

```diff
@@ -215,8 +250,37 @@ class common {
                 println color("uncommitted changes. Skipping.", Ansi.YELLOW)
             } else {
                 println color("updating $itemType $itemName", Ansi.GREEN)
+                File targetDirFetchHead = new File("$targetDir/.git/FETCH_HEAD")
+                def timeLimit = use(groovy.time.TimeCategory){ 10.minute }
+                if (skipRecentUpdates && targetDirFetchHead.exists() && isRecentlyUpdated(targetDirFetchHead, timeLimit)){
+                    println color("Skipping update for $itemName: updated within last $timeLimit", Ansi.YELLOW)
+                    return
+                }
+                if (targetDirFetchHead.exists()){
+                    targetDirFetchHead.setLastModified(new Date().getTime())
+                }
```

Comment:
This should at least be extracted to a method for better readability.

I think this bit clutters our code, and in my opinion for no apparent reason other than "because it is technically possible". We are not using Github API to worry about rate limiting, so I don't understand why we would ever need this feature. 

@Cervator I'd rather have 20-30 lines of code less to maintain (and/or go stale)...

Reply:
That's the main bit of hesitancy I've got now that my original use case doesn't result in the update halting early. If that still happened in mega-workspaces (I've hit that regularly in the past, having various GCI modules etc) then I'd still go for this for sure. But now, that I'm not sure it'll ever be needed? Conflicted 😕 

To be clear the time it takes to chug through a 200 module megaworkspace with an update-all possibly several times (if something breaks) was the motivation, less so the stress on GitHub. Although my digital treehugger self does lose a tiny bit of sleep at night over that as well :-)

Reply:
I didn't fully extract to a new method, but did make it more readable. It was looking to me like extracting wouldn't make it much more readable, let me know what you think